packages/ipfs-daemon/src/create-ipfs.ts

```diff
@@ -118,54 +113,98 @@ async function createIpfsOptions(
   )
 }
 
-const createInstanceByType = {
-  go: async (ipfsOptions: Ctl.IPFSOptions, disposable = true): Promise<IpfsApi> => {
-    if (!ipfsOptions.start) {
-      throw Error('go IPFS instances must be started')
-    }
-    const ipfsd = await createController(ipfsOptions, disposable)
-    // API is only set on started controllers
-    const started = await ipfsd.start()
-    return started.api
-  },
+export type GoIpfsFlavor = {
+  name: 'go'
+  options: IPFSOptions
+}
+
+export type RustIpfsFlavor = {
+  name: 'rust'
+  options: RustIpfsOptions
 }
+
+type IpfsFlavor = GoIpfsFlavor | RustIpfsFlavor
+
+export async function createIPFS(goOptions: IPFSOptions = {}, disposable = true, rustOptions: RustIpfsOptions = { type: 'binary' } as RustIpfsOptions): Promise<IpfsApi> {
+  const env_flavor = process.env.IPFS_FLAVOR || 'go'
+  if (env_flavor == 'go') {
+    return (
+      await createIPFSFlavor(
+        {
+          name: 'go',
+          options: goOptions,
+        } as GoIpfsFlavor,
+        disposable
+      )
+    ).api()
+  } else {
+    return (
+      await createIPFSFlavor(
+        {
+          name: 'rust',
+          options: rustOptions,
+        } as RustIpfsFlavor,
+        disposable
+      )
+    ).api()
+  }
+}
+
+export interface RunningIpfs {
+  api(): IpfsApi
+
+  shutdown(logger?: DiagnosticsLogger): Promise<void>
+}
+
 /**
  * Create an IPFS instance
- * @param overrideConfig - IFPS config for override
+ * @param flavor - IPFS flavor to create
+ * @param disposable - ???
```

Comment:
Should this be filled in?

Reply:
Not sure what this is, @dbcfd Can you add some clarity? I remember looking at what disposable did and now I don't remember.

Reply:
I think it means when done with the ipfs to shut it down. If not disposable, it leaves it up. I think it does this with proxy to stop or not.